podman_hpc/hook_tool.py

```diff
@@ -1,209 +1,364 @@
 #!/usr/bin/python3
+"""
+OCI hook tool for Podman-HPC.
+
+This module provides utilities used by the OCI hook to:
+- Enter a target process namespace
+- Copy or bind-mount files and directories into the container root
+- Load module configurations and apply rules
+"""
 import ctypes
 import ctypes.util
 import os
 import sys
-import errno
 import subprocess
 import json
-import yaml
 import shutil
 import re
 from glob import glob, iglob
+from typing import Dict, Callable, Iterable
 
 _MOD_ENV = "PODMANHPC_MODULES_DIR"
 
 _libc = ctypes.CDLL(ctypes.util.find_library("c"), use_errno=True)
-_libc.mount.argtypes = (
-    ctypes.c_char_p,
-    ctypes.c_char_p,
-    ctypes.c_char_p,
-    ctypes.c_ulong,
-    ctypes.c_char_p,
-)
+# Ensure ctypes call signature for setns is specified for safer FFI calls
+try:
+    _libc.setns.argtypes = (ctypes.c_int, ctypes.c_int)
+except AttributeError:
+    # Some libc stubs may not expose setns; we will fail later on use
+    pass
 logger = None
 
 
-def log(msg):
+def log(msg: str) -> None:
+    """
+    Write a line to the configured log stream if present.
+    """
     if logger:
         logger.write(f"{msg}\n")
+        try:
+            logger.flush()
+        except Exception:
+            pass
 
 
-def setns(pid, ns):
+def setns(pid: int, ns: str) -> None:
     """
-    Attach to a namespace using the pid
-    ns = ["mnt", "pid", "net", ...]
+    Attach to a namespace of a target process ID.
+
+    Parameters
+    - pid: Process ID whose namespace to join.
+    - ns: Name of the namespace, e.g. "mnt", "pid", "net", ...
+
+    Raises
+    - OSError: if attaching to the namespace fails.
     """
-    pth = f"/proc/{pid}/ns/{ns}"
-    tgt = open(pth, "r")
```

Comment:
old version doesn't use context mgr - is not closing the file when this func goes out of scope intended?